### ISSUE TITLE
Nginx conf for pydio

### ISSUE THREAD

**Issue by studioai** [MichaIng/DietPi#297]
I wrote a nginx conf file for pydio!

pydio works fine with it in /etc/nignx/sites-dietpi, even webdav and file share link redirection are OK!

So, Please. Apply this file to next dietpi release.

Then, I'll be happy for my little contribution to this wonderful project.

`I wrote a nginx conf file for pydio!

pydio works fine with it in /etc/nignx/sites-dietpi, even webdav and file share link redirection are OK!

So, Please. Apply this file to next dietpi release.

Then, I'll be happy for my little contribution to this wonderful project.

```
        rewrite ^/pydio/dashboard|^/pydio/settings|^/pydio/welcome|^/ws- /pydio/index.php last;

   if ( !-e $request_filename ) {
      # WebDAV Rewrites
      rewrite ^/pydio/shares /pydio/dav.php last;
      # Sync client
      rewrite ^/pydio/api /pydio/rest.php last;
      # External users 
      rewrite ^/pydio/user /pydio/index.php?get_action=user_access_point last;
      # Public shares
      rewrite ^/pydio/data/public/([a-zA-Z0-9_-]+)\.php$ /pydio/data/public/share.php?hash=$1?;
   }
   rewrite ^/pydio/data/public/([a-zA-Z0-9_-]+)--([a-z]+)$ /pydio/data/public/share.php?hash=$1&lang=$2?;
   rewrite ^/pydio/data/public/([a-zA-Z0-9_-]+)$ /pydio/data/public/share.php?hash=$1?;

   # Prevent Clickjacking
   add_header X-Frame-Options "SAMEORIGIN";

   # Only allow these request methods and do not accept DELETE, SEARCH and other methods
   if ( $request_method !~ ^(GET|HEAD|POST|PROPFIND|OPTIONS)$ ) {
      return 444;
   }

   location ~* ^/pydio/(?:\.|conf|data/(?:files|personal|logs|plugins|tmp|cache)|plugins/editor.zoho/agent/files) {
      deny all;
   }

   # Enables Caching
   location ~* \.(ico|css|js)$ {
      expires 7d;
      add_header Pragma public;
      add_header Cache-Control "public, must-revalidate, proxy-revalidate";
   }
```

**Comment by Fourdee** [comment]
@studioai 
Thanks for this, although could you please explain the differences in your configuration file and its benefits, when compared to our existing one?

**Comment by studioai** [comment]
As far I know there's no existing one for nginx. Only apache configuration file is given for pydio installation. So, I googled for nginx configuration for pydio and modified it.

It is automatically included if it's in /etc/nignx/sites-dietpi directory with .config extention by default nginx configuration file.

this is for who installed LEMP (NginX + MySQL + Php) Stack and Installed Pydio later.

**Comment by Fourdee** [comment]
@studioai 
As of DietPi version 115, all the `dietpi-software` webbased installations now support Lighttpd, Nginx and apache2 stacks. Rob did alot of work on getting the Nginx and Lighttpd installations and configurations working.

What version are you running?

**Comment by studioai** [comment]
@Fourdee 
I have 116 Installed now (It was 115 image when I installed)

There was /etc/nginx/sites-enabled/default only. Am I missing something?

If I'm not, default configuration supports php well. but doen't go well with Pydio's WebDav service and File Sharing via URL because there is no proper rewrite options.

What I wrote is additional configuration file can be add on to default conf. and that defines rewrite rules for pydio. It doesn't touch anything on default configuration. Only It need to be in /etc/nginx/sites-dietpi directory. Then it will be included by default configuration. No harm.

It does same rule as [this configration file](https://github.com/Fourdee/DietPi/blob/master/dietpi/conf/nginx.sites-dietpi.owncloud.conf) for pydio
